lib/stdlib/src/binary.erl

```diff
@@ -934,6 +935,31 @@ unhex(X) ->
              no, 10, 11, 12, 13, 14, 15, no, no, no, no, no, no, no, no, no  %96
             }).
 
+-doc """
+Joins a list of binaries together by a specified `Separator`.
+
+Equivalent to `iolist_to_binary(lists:join(Separator, Binaries))` with `binary:join/2` being the faster of the two.
+
+_Example:_
+
+```erlang
+1> binary:join([<<"a">>, <<"b">>, <<"c">>], <<", ">>).
+<<"a, b, c">>
+```
+""".
+-doc(#{since => <<"OTP 27.0">>}).
+-spec join([binary()], binary()) -> binary().
+join([], _Separator) -> <<>>;
+join([H], _Separator) -> H;
+join([H | T], Separator) ->
+    Acc = <<>>,
+    join(T, Separator, <<Acc/binary, H/binary>>).
```

Comment:
```suggestion
join([H | T], Separator) ->
    join(T, Separator, H).
```

Reply:
@Maria-12648430 This change was specifically done based on earlier discussions with @michalmuskala 👍 It's an optimisation that I was at least not aware of but seems to make a nice difference. https://github.com/erlang/otp/pull/8100#pullrequestreview-1869763175

Reply:
Ok, news to me :sweat_smile: Never mind then :+1: